src/components/Button/Button.stories.tsx

```diff
@@ -0,0 +1,352 @@
+import ButtonGroup from "../ButtonGroup/ButtonGroup";
+import Icon from "../Icons/Icon";
+import Table from "../Table/Table";
+
+const iconNames = [
+  "accessibilityFull",
+  "accessibilityPartial",
+  "actionCheckCircle",
+  "actionCheckCircleFilled",
+  "actionExit",
+  "actionHelpDefault",
+  "actionHelpOutline",
+  "alertNotificationImportant",
+  "actionLaunch",
+  "actionRegistration",
+  "actionPower",
+  "actionSettings",
+  "alertNotificationImportant",
+  "alertWarningFilled",
+  "alertWarningOutline",
+  "arrow",
+  "building",
+  "check",
+  "clock",
+  "close",
+  "decorativeEnvelope",
+  "decorativeLibraryCard",
+  "decorativeShoppingBag",
+  "download",
+  "errorFilled",
+  "errorOutline",
+  "fileTypeAudio",
+  "fileTypeDoc",
+  "fileTypeGenericDoc",
+  "fileTypeImage",
+  "fileTypePdf",
+  "fileTypeSpreadsheet",
+  "fileTypeVideo",
+  "headset",
+  "legacyAccountFilled",
+  "legacyAccountUnfilled",
+  "legacySocialFacebook",
+  "legacySocialInstagram",
+  "legacySocialTwitter",
+  "legacySocialYoutube",
+  "locator",
+  "minus",
+  "plus",
+  "search",
+  "socialFacebook",
+  "socialInstagram",
+  "socialTumblr",
+  "socialTwitter",
+  "socialYoutube",
+  "speakerNotes",
+  "utilityAccountFilled",
+  "utilityAccountUnfilled",
+  "utilityHamburger",
+  "utilitySearch",
+];
+
+const meta: Meta<typeof Button> = {
+  title: "Components/Form Elements/Button",
+  component: Button,
+  decorators: [withDesign],
+  argTypes: {
+    buttonText: { description: "Only used for Storybook" },
+    buttonType: {
+      table: { defaultValue: { summary: "primary" } },
+    },
+    className: { control: false },
+    displayIcon: { description: "Only used for Storybook." },
+    displayIconLeft: {
+      description: "Move the icon left or right. Only used for Storybook.",
+    },
+    iconType: {
+      control: { type: "select" },
+      options: iconNames,
+      description: "Select the icon. Only used for Storybook.",
+    },
+    id: { control: false },
+    isDisabled: { table: { defaultValue: { summary: false } } },
+    mouseDown: { table: { defaultValue: { summary: false } } },
+    onClick: { control: false },
+  },
+};
+
+export default meta;
+type Story = StoryObj<typeof Button>;
+
+/**
+ * Main Story for the Button component. This must contains the `args`
+ * and `parameters` properties in this object.
+ */
+export const WithControls: Story = {
+  args: {
+    buttonText: "Button Text",
+    buttonType: "primary",
+    className: undefined,
+    displayIcon: false,
+    displayIconLeft: true,
+    iconType: "search",
+    id: "button-id",
+    isDisabled: false,
+    mouseDown: undefined,
+    onClick: undefined,
+    size: "medium",
+    type: "button",
+  },
+  parameters: {
+    design: {
+      type: "figma",
+      url: "https://www.figma.com/file/qShodlfNCJHb8n03IFyApM/Master?node-id=11477%3A2298",
+    },
+    jest: ["Button.test.tsx"],
+  },
+  render: () => <Button id="button-id">Button Text</Button>,
```

Comment:
Controls is not setup. Update this to:
```
render: (args) => <Button {...args}>{args.ButtonText}</Button>,
```
The original one renders icons and can place them on the left or right side. Should we keep that? If so, this may need to be updated to:
```
<Button
        buttonType={args.buttonType}
        className={args.className}
        id={args.id}
        isDisabled={args.isDisabled}
        mouseDown={args.mouseDown}
        onClick={args.onClick}
        size={args.size}
        type={args.type}
      >
        {args.displayIcon && args.displayIconLeft && (
          <Icon name={args.iconType} align="left" size="small" />
        )}
        {args.buttonText}
        {args.displayIcon && !args.displayIconLeft && (
          <Icon name={args.iconType} align="right" size="small" />
        )}
      </Button>
```